public/js/controllers/game.js

```diff
@@ -229,7 +255,8 @@ angular.module('mean.system')
   }
 
   $scope.searchUsers = function () {
-    $http.get(`/api/search/users/${$scope.email}`)
+    const link = `/api/search/users/${$scope.email}`;
+    dataFactory.searchUsers(link)
```

Comment:
Why have the `dataFactory.searchUsers` service in the first place then? You could simply just have the $http service in the controllers make the requests directly to `link`. But since you already created a `dataFactory`(which I think is a great idea), then you should follow through. (Don't blame me, you started this :bowtie:)

Ideally, you want to separate concerns as much as possible. The idea of moving the $http service to another service is to ensure that the controller isn't doing too much, make it more light weight and easier to test.

So yes, for every endpoint, you want to have a separate service method handle requests to the endpoint - Think Single Responsibility!